lib/client.js

```diff
@@ -170,23 +211,48 @@ Client.prototype._invoke = function(method, args, location, callback, options, e
   self.lastMessage = message;
   self.lastRequest = xml;
 
-  req = http.request(location, xml, function(err, response, body) {
+  self.emit('message', message);
+  self.emit('request', xml);
+
+  req = self.httpClient.request(location, xml, function(err, response, body) {
     var result;
     var obj;
     self.lastResponse = body;
     self.lastResponseHeaders = response && response.headers;
+    self.emit('response', body);
 
     if (err) {
       callback(err);
-    } else if (response.statusCode !== 200) {
-      callback(new Error('Invalid response: ' + response.statusCode + '\nBody: ' + body));
     } else {
       try {
         obj = self.wsdl.xmlToObject(body);
+
+        if (!obj.Body && response.statusCode !== 200) {
```

Comment:
If I read `wsdl.js` correctly and assuming we are checking for existence of `Body`, the only way we can reach line 231 is when `root.Envelope` is falsy. Otherwise, it would have thrown an exception at https://github.com/shelbys/node-soap/blob/node_0_12/lib/wsdl.js#L1390. So we are checking if `root` has `Body`. And if not, and response status is not `200`, we will callback with error. What exactly do we want to check here?